pkg/kubelet/apis/config/validation/validation_reserved_memory.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	corev1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+)
+
+// validateReservedMemory validates the reserved memory configuration and returns an error if it is invalid.
+func validateReservedMemoryConfiguration(kc *kubeletconfig.KubeletConfiguration) []error {
+	if len(kc.ReservedMemory) == 0 {
+		return nil
+	}
+
+	var errors []error
+
+	numaTypeDuplicates := map[int32]map[v1.ResourceName]bool{}
+	for _, reservedMemory := range kc.ReservedMemory {
+		numaNode := reservedMemory.NumaNode
+		if _, ok := numaTypeDuplicates[numaNode]; !ok {
+			numaTypeDuplicates[numaNode] = map[v1.ResourceName]bool{}
+		}
+
+		for resourceName, q := range reservedMemory.Limits {
+			if !reservedMemorySupportedLimit(resourceName) {
+				errors = append(errors, fmt.Errorf("the reserved memory has unsupported limit type %q for the NUMA node %d", resourceName, numaNode))
```

Comment:
`the limit type %q for NUMA node %d is not supported, only %v is accepted`, and for %v pass a strict array of the allowed values (hugepages and memory)